lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_order.rb

```diff
@@ -221,23 +223,27 @@ def tax_rate_amount(query_result_row)
         end
 
         def total_excl_tax(query_result_row)
+          order = order(query_result_row)
           amount = Spree::Adjustment.enterprise_fee
-            .where(order: order(query_result_row))
+            .where(order:)
             .where(originator_id: enterprise_fee_id(query_result_row))
             .pick("sum(amount)") || 0
-          amount - tax(query_result_row, all: true, included: true)
+          total_excl_tax_amount = amount - tax(query_result_row, all: true, included: true)
+          apply_voucher_on_amount(order, total_excl_tax_amount)
         end
 
         def tax(query_result_row, all: false, included: nil)
-          order_id = order(query_result_row).id
+          order = order(query_result_row)
+          order_id = order.id
           adjustment_ids = enterprise_fee_adjustment_ids(query_result_row)
           query = Spree::Adjustment.tax
           query = query.where(included: true) unless included.nil?
           query = query.where(originator_id: tax_rate_id(query_result_row)) unless all == true
-          query.where(order_id:)
+          tax_amount = query.where(order_id:)
```

Comment:
We could actually remove the `order_id` variable and refer directly to `order`. AR automatically uses the ID anyway.

Reply:
That's great! :heart_eyes: 

It's updated here: https://github.com/openfoodfoundation/openfoodnetwork/pull/11985/commits/1d44a3ffe04de6faae91a4aeb546122dd57320b4